snappy/snapdata.go

```diff
@@ -45,6 +45,23 @@ func RemoveSnapData(snap *snap.Info) error {
 	return nil
 }
 
+// RemoveSnapSharedData removes the data shared between versions of the given snap
+func RemoveSnapSharedData(snap *snap.Info) error {
+	dirs, err := snapSharedDataDirs(snap)
+	if err != nil {
+		return err
+	}
+
+	for _, dir := range dirs {
+		if err := os.RemoveAll(dir); err != nil && !os.IsNotExist(err) {
+			return err
+		}
+		os.Remove(filepath.Dir(dir))
```

Comment:
Based on
```
zyga@zyga-thinkpad-w510:~$ go doc os.RemoveAll
func RemoveAll(path string) error
    RemoveAll removes path and any children it contains. It removes everything
    it can but returns the first error it encounters. If the path does not
    exist, RemoveAll returns nil (no error).
```

I don't think you have to check for the `os.IsNotExist` or call `os.Remove` again later.